### PR TITLE
Hierarchies-react: Update README to use compiled code snippets

### PR DESCRIPTION
**Pull request by jasdom**
Closes https://github.com/iTwin/presentation/issues/666.

### CODE REVIEW

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      });
+
+      afterEach(async () => {
+        await terminate();
+      });
+
+      it("Tree", async function () {
+        // set up imodel for the test
+        // eslint-disable-next-line deprecation/deprecation
+        const imodel = await buildTestIModel(this, (builder) => {
```
[the remark this answers is not in the document]
Comment (jasdom) [reply apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx]: Changed.

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+// __PUBLISH_EXTRACT_END__
+
+describe("Hierarchies-react", () => {
+  describe("Learning snippets", () => {
+    describe("Readme example", () => {
+      beforeEach(async () => {
+        await initialize();
+        RpcManager.registerImpl(ECSchemaRpcInterface, ECSchemaRpcImpl);
+        RpcConfiguration.developmentMode = true;
+        RpcManager.initializeInterface(ECSchemaRpcInterface);
```
[the remark this answers is not in the document]
Comment (jasdom) [reply apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx]: Changed.

File: packages/hierarchies-react/README.md [hunk 3]
```diff
@@ -88,29 +111,54 @@ The returned result is identical to the one from [`useTree`](#usetree).
...
-}
+<!-- END EXTRACTION -->
+
+After providing the unified selection storage through the context, the `useUnifiedSelectionTree` hook can be used inside the component as follows:
+
+<!-- [[include: Presentation.HierarchiesReact.Readme.UseUnifiedSelectionTree, tsx]] -->
+<!-- BEGIN EXTRACTION -->
+
+```tsx
+const { rootNodes, ...state } = useUnifiedSelectionTree({
```
[the remark this answers is not in the document]
Comment (jasdom) [reply packages/hierarchies-react/README.md]: Added a separate piece of code in `ReadmeExample.test.tsx`.

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            ecsql: `
+              SELECT
+                ${await nodesQueryFactory.createSelectClause({
+                  ecClassId: { selector: "this.ECClassId" },
+                  ecInstanceId: { selector: "this.ECInstanceId" },
+                  nodeLabel: {
+                    selector: await labelsQueryFactory.createSelectClause({ classAlias: "this", className: "BisCore.PhysicalModel" }),
+                  },
+                  hasChildren: false,
+                  hideIfNoChildren: false,
```
[the remark this answers is not in the document]
Comment (jasdom) [reply apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx]: Forgot to remove, fixed.

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+  const { rootNodes, isLoading, ...state } = useUnifiedSelectionTree({
+    // the source name is used to distinguish selection changes being made by different components
+    sourceName: "MyTreeComponent",
+    // the iModel key is required for unified selection system to distinguish selection changes between different iModels
+    imodelKey,
+    // iModel access is used to build the hierarchy
+    imodelAccess,
+    // the hierarchy definition describes the hierarchy using ECSQL queries; here it just returns all bis.Model instances
+    // grouped by class
```
[the remark this answers is not in the document]
Comment (jasdom) [reply apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx]: Updated.

File: packages/hierarchies-react/README.md [hunk 6]
```diff
@@ -88,29 +111,54 @@ The returned result is identical to the one from [`useTree`](#usetree).
...
+
 ```tsx
-import { UnifiedSelectionProvider, useUnifiedSelectionTree } from "@itwin/presentation-hierarchies-react";
-import { createStorage } from "@itwin/unified-selection";
+import { TreeRenderer, UnifiedSelectionProvider, useUnifiedSelectionTree } from "@itwin/presentation-hierarchies-react";
+import { useEffect, useState } from "react";
 
-// Unified selection storage should be created once per application and reused across components.
+// Not part of the package - this should be created once and reused across different components of the application.
 const selectionStorage = createStorage();
```
[the remark this answers is not in the document]
Comment (jasdom) [reply packages/hierarchies-react/README.md:122]: Moved the import into correct snippet.

File: packages/hierarchies-react/README.md [hunk 7]
```diff
@@ -14,24 +14,47 @@ It takes 2 required properties:
...
-    const schemaProvider = createECSchemaProvider(getIModelSchemaContext(imodel));
-    return {
-      ...schemaProvider,
-      // while caching for hierarchy inspector is not mandatory, it's recommended to use it to improve performance
-      ...createCachingECClassHierarchyInspector({ schemaProvider, cacheSize: 100 }),
-      // the second argument is the maximum number of rows the executor will return - this allows us to
-      // avoid creating hierarchy levels of insane size (expensive to us and useless to users)
-      ...createLimitingECSqlQueryExecutor(createECSqlQueryExecutor(imodel), 1000),
-    };
+<!-- [[include: [Presentation.HierarchiesReact.iModelAccess.Imports, Presentation.HierarchiesReact.iModelAccess], tsx]] -->
```
[the remark this answers is not in the document]
Comment (jasdom) [reply packages/hierarchies-react/README.md]: Indented.

File: packages/hierarchies-react/README.md [hunk 8]
```diff
@@ -213,51 +267,64 @@ function MyTreeComponentInternal({ imodelAccess, imodelKey }: { imodelAccess: IM
...
+
 ```tsx
+import {
+  createRenderedTreeNodeData,
+  LocalizationContextProvider,
+  RenderedTreeNode,
+  TreeNodeRenderer,
+  TreeRenderer,
+  useUnifiedSelectionTree,
+} from "@itwin/presentation-hierarchies-react";
```
[the remark this answers is not in the document]
Comment (jasdom) [reply packages/hierarchies-react/README.md]: Split the imports.

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/Localization.test.tsx [hunk 9]
```diff
@@ -4,38 +4,39 @@
...
   RenderedTreeNode,
   TreeNodeRenderer,
   TreeRenderer,
-  useUnifiedSelectionTree,
 } from "@itwin/presentation-hierarchies-react";
-
 // __PUBLISH_EXTRACT_END__
 
-describe("Hierarchies-react", () => {
+describe.only("Hierarchies-react", () => {
```
Comment (grigasp) [inline apps/full-stack-tests/src/hierarchies-react/learning-snipptets/Localization.test.tsx]:
```suggestion
describe("Hierarchies React", () => {
```

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx [hunk 10]
```diff
@@ -6,154 +6,178 @@
...
+      ...createLimitingECSqlQueryExecutor(createECSqlQueryExecutor(imodel), 1000),
+    };
   }
-  return <TreeRenderer {...state} rootNodes={rootNodes ?? []} />;
-}
-// __PUBLISH_EXTRACT_END__
+  // __PUBLISH_EXTRACT_END__
 
-describe("Hierarchies-react", () => {
+describe.only("Hierarchies-react", () => {
```
Comment (grigasp) [inline apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx]:
```suggestion
describe("Hierarchies React", () => {
```

**Review by grigasp — APPROVED** [review]

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/Localization.test.tsx (hunk 9 above)
Comment (jasdom) [reply apps/full-stack-tests/src/hierarchies-react/learning-snipptets/Localization.test.tsx]: I think you were looking at outdated files as the `.only` was already removed. Changed the name though.

File: apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx (hunk 10 above)
Comment (jasdom) [reply apps/full-stack-tests/src/hierarchies-react/learning-snipptets/ReadmeExample.test.tsx]: Changed.

**Pull request merged**